File: Quid_pro_Quo/WebApiModels/PostFormApiModel.cs
using Microsoft.AspNetCore.Http;

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Quid_pro_Quo.WebApiModels
{
    public class PostFormApiModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public IEnumerable<IFormFile> ImageFiles { get; set; }

        public IEnumerable<DateTime> PerformServiceOnDatesList { get; set; }
        public string PerformServiceInPlace { get; set; }
    }
}

File: Quid_pro_Quo/ClientApp/src/app/models/post-form-api.model.ts
export interface PostFormApiModel {
  id: number;
  title: string;
  text: string;
  imageFiles: Blob[];
  performServiceOnDatesList: Date[];
  performServiceInPlaceLat: number;
  performServiceInPlaceLng: number;
  performServiceInPlaceZoom: number;
}
